### PR TITLE
Fixed #697: Differentiate superimposed stops and remove directions in route map

### LINKED ISSUE #697: Differentiate superimposed stops and remove directions in route map
### Expected Behaviour
- When more than one client are at the same stop (for example many addresses are in the same building) such that the alphabetic letter labels are superimposed on the map, find a way to indicate this on the map or on the sequence list that accompanies the map.
- The label text A=santropol should be changed to A=Santropol Roulant.
- There should be an extra stop at the end of the sequence such that the delivery volunteer sees the return path from the last client in the route back to the Santropol Roulant.
- Remove the "Directions" in the lower right section of the web page.
- Remove the "Directions" on the printed map.

### Steps to reproduce
- Make sure that the data contains, for a given geolocated address, more than one client (the only difference being the apartment number)
- Run the Kitchen Count steps 1, 2, 3 and 4
- Click "Organize" for a non empty route

### PR DESCRIPTION
**Pull request by peakwinter**
### Changes proposed in this pull request:

On both the route planner and printout...

* ...multiple "waypoints" (clients) at the same address show on the map as a special marker with a star in it, indicating that more than one client can be found at the same location.
* ...when an address has multiple clients, the address is only listed once, and the clients' names are comma-separated.
* ...a final waypoint has been added, showing the itinerary for the return trip to Santropol Roulant.
* ...the turn-by-turn directions have been removed.

### Status

- [X] READY
- [ ] HOLD
- [ ] WIP (Work-In-Progress)

### How to verify this change

* Ensure that you have at least two clients in a single route that share a street address and a delivery on the date in question.
* Proceed through the Kitchen Count process until you arrive at the Routes Information screen.
* Browse to the route that has the two clients in it, and click Organize.
* Verify that:
  * ... one map marker with a white star appears above the street address that has two clients
  * ... the waypoint list in the sidebar contains only one entry for said address, and both clients' names show up as comma-separated.
  * ... the turn-by-turn directions do not appear
* Click "Map and Directions" and repeat the aforementioned verifications.

### Deployment notes and migration

- [ ] Migration is needed for this change

### New translatable strings

- [ ] If applicable, I have included updated .po files with new strings (see https://github.com/savoirfairelinux/sous-chef/blob/dev/docs/internationalization.adoc)

### Additional notes

* The mechanism to group map markers (and waypoints) by address is done by comparing the latitude and longitude of the geocoded points for the clients in question. If the latitudes and longitudes are equal to within five decimal points of tolerance, the clients will be considered to live at the same street address.
* Apartment numbers are not listed on the Delivery Route printout, rather only on the Routes Information printout. I did not change this as it was existing behaviour and the issue did not request a change in this regard.

### CODE REVIEW

**Comment by peakwinter** [comment]
This PR is ready for review.

**Comment by lingxiaoyang** [comment]
Behaviors (copied from issue):

- [ ] When more than one client are at the same stop (for example many addresses are in the same building) such that the alphabetic letter labels are superimposed on the map, find a way to indicate this on the map or on the sequence list that accompanies the map.
This is marked with a "star" (@lamontfr is this ok?) and the popup shows all people and their address.
image
And in the print sheet, apartment # is not listed. @lamontfr  Is this the expected behavior?
image
- [ ] The label text A=santropol should be changed to A=Santropol Roulant.
In the list this is changed. But in the popup it says "Santro". @peakwinter 
image
- [x] There should be an extra stop at the end of the sequence such that the delivery volunteer sees the return path from the last client in the route back to the Santropol Roulant.
Verified.
- [x] Remove the "Directions" in the lower right section of the web page.
Verified.
- [x] Remove the "Directions" on the printed map.
Verified,

File: src/frontend/js/custom-leaflet.js [hunk 1]
```diff
@@ -183,7 +222,7 @@ function main_map_init (map, options) {
                 var marker;
 
                 // adjust marker according waypoints
-                if (wp.name == "santropol") {
+                if (wp.santro && !wp.hideMarker) {
                     // add awesome marker for santropol
                     marker =  L.marker([45.516564, -73.575145], {
                         draggable: false,
```
Comment (lingxiaoyang) [inline src/frontend/js/custom-leaflet.js]: A question here (not mentioned in the issue but I think is better to do this): can we also set other markers with "draggable:false"?

Comment (peakwinter) [inline src/frontend/js/custom-leaflet.js]: This was explicitly added in the file in several locations, but I am also not sure if it is really necessary. In disabling draggable markers one can still create a custom itinerary by dragging the lines between markers.

Comment (lingxiaoyang) [inline src/frontend/js/custom-leaflet.js]: @peakwinter Really? We then keep it unchanged unless required.

**Comment by lamontfr** [comment]
@peakwinter @lingxiaoyang
Re: **Question 1**: Is a star is OK as a label when there are more than one client at the same stop ?
I think that a two-character label would be preferable for example "B*" because : 
- On a route we could have more than one stop for which there are many clients;
- The delivery persons really need the map to find their way around and so in the map screenshot above they would not find stop B on the map.

Re: **Question 2**: There is no apartment number in the "Route sequence" below the map. Is this OK ?
I think that this is correct, since : 
- Many clients may live in the same building;
- The detailed route sheet (printed separately from the map) includes the client's full address with Apartment number and telephone number, which are essential.

I will meet Jean-François tomorrow and check the above with him.

**Comment by lamontfr** [comment]
@peakwinter @lingxiaoyang I just finished a meeting with Jean-François and he confirmed my answers to your questions. So we are good to go.

**Comment by peakwinter** [comment]
Thanks @lamontfr !

This is therefore ready to be merged if the final review passes :) @lingxiaoyang

**Comment by lingxiaoyang** [comment]
@lamontfr @peakwinter  Thank you!

**Pull request merged**